dspace-api/src/main/java/org/dspace/discovery/IndexEventConsumer.java

```diff
@@ -151,37 +151,37 @@ public void consume(Context ctx, Event event) throws Exception {
     @Override
     public void end(Context ctx) throws Exception {
 
-        if (objectsToUpdate != null && handlesToDelete != null) {
+        if (objectsToUpdate != null && uniqueIdsToDelete != null) {
 
             // update the changed Items not deleted because they were on create list
-            for (DSpaceObject o : objectsToUpdate) {
+            for (IndexableObject iu : objectsToUpdate) {
                 /* we let all types through here and
                  * allow the search indexer to make
                  * decisions on indexing and/or removal
                  */
-                DSpaceObject iu = ctx.reloadEntity(o);
-                String hdl = iu.getHandle();
-                if (hdl != null && !handlesToDelete.contains(hdl)) {
+                // iu = ctx.reloadEntity(o);
```

Comment:
Yes I think we still need this to prevent Hibernate issues in the future. In order to support this, you need to change the `IndexableObject` interface to `IndexableObject<PK extends Serializable> extends ReloadableEntity<PK>`.

WorkspaceItem and WorkflowItem already implement the `ReloadableEntity` interface

Reply:
yes, because we now use the unique id to delete the record instead than the handle and the IDs are always available in the object also if it is not completely initialized so it cannot raise a LazyInizializationException. I will remove the commented code

Reply:
I have seen the @tomdesair comment later. I agree that could be safer for the future to keep the reload so I'm going to introduce the modification suggested by @tomdesair

Reply:
done